src/Model/Image.php

```diff
@@ -21,14 +21,19 @@ final class SitemapImageUrl implements SitemapImageUrlInterface
     /** @var string|null */
     private $license;
 
+    public function __construct(string $location)
```

Comment:
Valid, but shall we then also remove the setters from the required properties? Don't think they should be used then anymore actually. Or maybe I'm overlooking an use-case.

Alternative is to try to get rid of all setters in all models anyway, not sure what's the impact of this.

Reply:
In the future we will probably dispatch events various places. In that situation a user of the plugin could add their own touch to any of the mutable properties, which I think is a good idea.

Reply:
Hm, I'm not sure I agree here. I think the mutability should happen in the locations where the factories are to be used (generators, etc.) and not allow it to change afterwards anymore. Let's file an issue to look into this, to get rid of setters where possible. Doesn't need to be in this PR.